src/main/java/junitparams/internal/ParameterisedTestMethodRunner.java

```diff
@@ -68,6 +69,14 @@ private Description findChildForParams(Statement methodInvoker, Description meth
     }
 
     private InvokeParameterisedMethod findParameterisedMethodInvokerInChain(Statement methodInvoker) {
+        if (methodInvoker != null && methodInvoker instanceof Fail) {
+            try {
+                methodInvoker.evaluate();
+            } catch (Throwable throwable) {
+                throw new IllegalStateException("Instantiation failed", throwable);
```

Comment:
I think calling `methodInvoker.evaluate()` would be enough for the fix. Ther's no need to wrap up the original cause in `IllegalStateException`. WDYT?

Reply:
The reason for that is that `evaluate` is throwing `Throwable` and thus method `findParametr...` will not compile.

Reply:
Well, there is better solution - just return the value to the calling function.

Reply:
Do you have plans to complete the PR by adding a missing test case for the fix ([as suggested by the maintainers](https://github.com/Pragmatists/JUnitParams/pull/144#issuecomment-345837836))?